document/execution/numerics.rst

```diff
@@ -1,54 +1,1472 @@
+
+* Else if both :math:`z_1` and :math:`z_2` are zeroes of equal sign, then return positive zero.
+
+* Else if both :math:`z_1` and :math:`z_2` are zeroes of opposite sign, then return :math:`z_1`.
+
+* Else if :math:`z_2` is a zero, then return :math:`z_1`.
+
+* Else if :math:`z_1` is a zero, then return :math:`z_2` negated.
+
+* Else if both :math:`z_1` and :math:`z_2` are the same value, then return positive zero.
+
+* Else return the result of subtracting :math:`z_2` from :math:`z_1`, :ref:`rounded <aux-ieee>` to the nearest representable value.
+
+.. math::
+   \begin{array}{@{}lcll}
+   \fsub_N(\pm \NAN(n), z_2) &=& \nan_N\{\pm \NAN(n), z_2\} \\
+   \fsub_N(z_1, \pm \NAN(n)) &=& \nan_N\{\pm \NAN(n), z_1\} \\
+   \fsub_N(\pm \infty, \pm \infty) &=& \nan_N\{\} \\
+   \fsub_N(\pm \infty, \mp \infty) &=& \pm \infty \\
+   \fsub_N(z_1, \pm \infty) &=& \mp \infty \\
+   \fsub_N(\pm \infty, z_2) &=& \pm \infty \\
+   \fsub_N(\pm 0, \pm 0) &=& +0 \\
+   \fsub_N(\pm 0, \mp 0) &=& \pm 0 \\
+   \fsub_N(z_1, \pm 0) &=& z_1 \\
+   \fsub_N(\pm 0, \pm q_2) &=& \mp q_2 \\
+   \fsub_N(\pm q, \pm q) &=& +0 \\
+   \fsub_N(z_1, z_2) &=& \ieee_N(z_1 - z_2) \\
+   \end{array}
+
+.. note::
+   Up to the non-determinism regarding NaNs, it always holds that :math:`\fsub_N(z_1, z_2) = \fadd_N(z_1, \fneg_N(z_2))`.
+
+
+.. _op-fmul:
+
+:math:`\fmul_N(z_1, z_2)`
+.........................
+
+* If either :math:`z_1` or :math:`z_2` is a NaN, then return an element of :math:`\nan_N\{z_1, z_2\}`.
+
+* Else if one of :math:`z_1` and :math:`z_2` is a zero and the other an infinity, then return an element of :math:`\nan_N\{z_1, z_2\}`.
+
+* Else if both :math:`z_1` and :math:`z_2` are infinities of equal sign, then return positive infinity.
+
+* Else if both :math:`z_1` and :math:`z_2` are infinities of opposite sign, then return negative infinity.
+
+* Else if one of :math:`z_1` or :math:`z_2` is an infinity and the other a value with equal sign, then return positive infinity.
+
+* Else if one of :math:`z_1` or :math:`z_2` is an infinity and the other a value with opposite sign, then return negative infinity.
+
+* Else return the result of multiplying :math:`z_1` and :math:`z_2`, :ref:`rounded <aux-ieee>` to the nearest representable value.
+
+.. math::
+   \begin{array}{@{}lcll}
+   \fmul_N(\pm \NAN(n), z_2) &=& \nan_N\{\pm \NAN(n), z_2\} \\
+   \fmul_N(z_1, \pm \NAN(n)) &=& \nan_N\{\pm \NAN(n), z_1\} \\
+   \fmul_N(\pm \infty, \pm 0) &=& \nan_N\{\} \\
+   \fmul_N(\pm \infty, \mp 0) &=& \nan_N\{\} \\
+   \fmul_N(\pm 0, \pm \infty) &=& \nan_N\{\} \\
+   \fmul_N(\pm 0, \mp \infty) &=& \nan_N\{\} \\
+   \fmul_N(\pm \infty, \pm \infty) &=& +\infty \\
+   \fmul_N(\pm \infty, \mp \infty) &=& -\infty \\
+   \fmul_N(\pm q_1, \pm \infty) &=& +\infty \\
+   \fmul_N(\pm q_1, \mp \infty) &=& -\infty \\
+   \fmul_N(\pm \infty, \pm q_2) &=& +\infty \\
+   \fmul_N(\pm \infty, \mp q_2) &=& -\infty \\
+   \fmul_N(z_1, z_2) &=& \ieee_N(z_1 \cdot z_2) \\
```

Comment:
Ah, I think in general rules +-0.0 * +-q have to be added, because sometimes the result is -0.0